### PR TITLE
[format -> prettyprint] Redirect (fixes #876)

### LINKED ISSUE #876: rm format
Once #36 is done, src/format should be replaced by src/prettyprint. src/format should then be deleted.

### PR DESCRIPTION
**Pull request by artagnon**
Fix extraction to use src/prettyprint instead of src/format, and remove
src/format completely, since it is redundant. (Fixes #876)

Signed-off-by: Ramkumar Ramachandra <artagnon@gmail.com>

### CODE REVIEW

**Comment by catalin-hritcu** [comment]
The latest CI error seems to be
```
https://everestlogstorage.blob.core.windows.net/fstar-linux/buildlogfile_FStar-Linux-20190430-19.html
```
Can you access this, by the way?
https://everestlogstorage.blob.core.windows.net/fstar-linux/buildlogfile_FStar-Linux-20190430-19.html

**Comment by artagnon** [comment]
Thanks for the build log -- yeah, I was able to access it. Fix was simple enough.

File: src/extraction/FStar.Extraction.Kremlin.fs [hunk 1]
```diff
@@ -25,7 +25,7 @@ open FStar.Util
 open FStar.Extraction
 open FStar.Extraction.ML
 open FStar.Extraction.ML.Syntax
-open FStar.Format
+open FStar.Pprint
```
Comment (victor-dumitrescu) [inline src/extraction/FStar.Extraction.Kremlin.fs]: Nit: This import could be remove altogether.

File: src/extraction/FStar.Extraction.ML.Code.fs [hunk 2]
```diff
@@ -60,6 +60,57 @@ let e_app_prio        = (10000, Infix Left)
...
+    combine break1 docs
+
+let nest (i : int) (Doc d) =
+    Doc (d)
+
+let align (docs : list<doc>) =
+    let (Doc doc) = combine hardline docs in
+    Doc (doc)
+
+let hbox (d : doc) = d (* FIXME *)
```
Comment (victor-dumitrescu) [inline src/extraction/FStar.Extraction.ML.Code.fs]: Nit: This seems to be moving all the helpers from FStar.Format. Some of them could be removed completely (e.g., `break_`, `break0`, `cat1`).

**Comment by artagnon** [comment]
Fixed.

**Comment by catalin-hritcu** [comment]
This CI failure has nothing to do with your PR, but more of a CI / VSTS issue:
```
Retrieve extra VSTS build information
Unhandled: Request error with code: '401'.
```

**Comment by nikswamy** [comment]
I would have expected a snapshot diff in src/ocaml-output. Did the printed output really not change?

**Comment by artagnon** [comment]
> I would have expected a snapshot diff in src/ocaml-output. Did the printed output really not change?

I thought our official policy was to not burden contributors with updates to `src/ocaml-output` -- that should be taken care of automatically by the CI? See [CONTRIBUTIONS.md](https://github.com/FStarLang/FStar/blob/master/CONTRIBUTIONS.md#snapshots)

**Comment by nikswamy** [comment]
@catalin-hritcu @artagnon: I don't think this should have been merged in its current form.

I realize that CONTRIBUTIONS.md says that we should let the CI system push a snapshot automatically. But, that's a mistake there. The CI system does not refresh snapshots automatically: it cannot, since master is locked. I think we've discussed this issue at the F* meetings, and @protz  had a suggestion to create another unlocked staging branch for snapshot refreshes.

Until we fix this issue, I think we need to explicitly kick snapshot refreshing builds on the CI system on a branch of your choice before merging to master. I've done that now on a separate branch and will merge it into master if/when I get a green.

Put another way, I don't think we should merge to master unless we have a full green from CI, unless we have a specific, good reason for why we want to temporarily introduce some partial breakage.

**Comment by catalin-hritcu** [comment]
Let's fix `CONTRIBUTIONS.md` then, we both did precisely what the documentation said we should do.

**Comment by nikswamy** [comment]
Yes, see a separate PR for the update to CONTRIBUTIONS.md

Not trying to blame you for a mistake or anything. But, the workflow here just revealed a bug in our process.

**Comment by artagnon** [comment]
> The CI system does not refresh snapshots automatically: it cannot, since master is locked.

Hm, what does dzomo do then? See 24e300200434 for instance.

**Pull request merged**